test/service/test_response.py

```diff
@@ -1743,6 +1743,133 @@ def test_pagination_search_after_search_before(self):
+                        'sampleEntityType': ['organoids'],
+                        'effectiveOrgan': ['Brain'],
+                        'id': [
+                            'Org_HPSI0214i-kucg_2_2',
+                            'Org_HPSI0214i-wibj_2_2',
+                            'Org_HPSI0314i-hoik_1_2',
+                            'Org_HPSI0314i-sojd_3_2',
+                        ],
+                        'modelOrgan': ['Brain'],
+                        'modelOrganPart': [None],
+                    }
+                ]
+            ],
+            'specimens': [
+                # A hit for a project with 2 different type samples
+                [
+                    {
+                        'sampleEntityType': ['cellLines'],
+                        'effectiveOrgan': ['immune system'],
+                        'id': ['Cell_line_2'],
+                        'cellLineType': ['primary'],
+                        'modelOrgan': ['immune system'],
+                    },
+                    {
+                        'sampleEntityType': ['specimens'],
+                        'effectiveOrgan': ['embryo'],
+                        'id': ['Specimen1'],
+                        'organ': ['embryo'],
+                        'organPart': ['skin epidermis'],
+                        'disease': ['normal'],
+                        'preservationMethod': [None],
+                        'source': ['specimen_from_organism'],
+                    },
+                ],
+                # A hit for a project with 1 sample
+                [
+                    {
+                        'sampleEntityType': ['specimens'],
+                        'effectiveOrgan': ['pancreas'],
+                        'id': ['DID_scRSq06_pancreas'],
+                        'organ': ['pancreas'],
+                        'organPart': ['islet of Langerhans'],
+                        'disease': ['normal'],
+                        'preservationMethod': [None],
+                        'source': ['specimen_from_organism'],
+                    }
+                ],
+            ],
+        }
+        url = self.base_url + '/index/projects'
+        for entity_type, expected_hits in expected_filter_hits.items():
+            with self.subTest(entity_type=entity_type):
+                params = {
+                    'filters': json.dumps({'sampleEntityType': {'is': [entity_type]}}),
+                    'catalog': self.catalog,
+                    'size': 5,
+                    'sort': 'projectTitle',
+                    'order': 'asc',
+                }
+                response = requests.get(url, params=params)
+                response.raise_for_status()
+                response_json = response.json()
+                hits = response_json['hits']
+                self.assertEqual(len(hits), len(expected_hits))
+                for hit, expected_samples in zip(hits, expected_hits):
+                    self.assertEqual(hit['samples'], expected_samples)
```

Comment:
I wonder if, instead, you should assert the entire hit. This would be stronger than just asserting the sample part, and would make the length assert redundant.

Reply:
I feel that this test is specific to the 'samples' inner entities and should keep the focus there.

Reply:
Fingerprinting might be a good compromise. Fingerprinting is when you derive a data structure (typically JSON) from the actual response and compare it against a literal, the expected data structure. That's my own invented term, BTW.